src/Versions.jl

```diff
@@ -8,34 +8,46 @@ export VersionBound, VersionRange, VersionSpec, semver_spec, isjoinable
 # VersionBound #
 ################
 struct VersionBound
-    t::NTuple{3,UInt32}
+    t::NTuple{4,UInt32}
     n::Int
     function VersionBound(tin::NTuple{n,Integer}) where n
-        n <= 3 || throw(ArgumentError("VersionBound: you can only specify major, minor and patch versions"))
-        n == 0 && return new((0,           0,      0), n)
-        n == 1 && return new((tin[1],      0,      0), n)
-        n == 2 && return new((tin[1], tin[2],      0), n)
-        n == 3 && return new((tin[1], tin[2], tin[3]), n)
+        n <= 4 || throw(ArgumentError("VersionBound: you can only specify major, minor, patch versions, and a single positive-integer-valued build number"))
```

Comment:
I made an error - of course we support zero as a build number.

```suggestion
        n <= 4 || throw(ArgumentError("VersionBound: you can only specify major, minor, patch versions, and a single non-negative-integer-valued build number"))
```